opm/simulators/linalg/bda/WellContributions.hpp

```diff
@@ -119,11 +121,19 @@ class WellContributions
     void freeStandardWells();
 
 public:
-
+#if HAVE_CUDA
     /// Set a cudaStream to be used
     /// \param[in] stream           the cudaStream that is used to launch the kernel in
-#if HAVE_CUDA
     void setCudaStream(cudaStream_t stream);
+    
+    /// Apply all Wells in this object
+    /// performs y -= (C^T * (D^-1 * (B*x))) for all Wells
+    /// \param[in] d_x        vector x, must be on GPU
+    /// \param[inout] d_y     vector y, must be on GPU
+    void apply(double *d_x, double *d_y);
+#elif HAVE_OPENCL
```

Comment:
As @Tongdongq already pointed out we can have CUDA and OpenCL. Hence please change to 
```
#endif 
#if HAVE_OPENCL
```
it is also not consistent with the [implementation](/OPM/opm-simulators/blob/be7fed413fc7e82d5639d40e854148c48def4997/opm/simulators/linalg/bda/WellContributions.cpp#L90-L132) and hence produces compile errors:
```
/home/mblatt/src/dune/opm-gpu/opm-simulators/opm/simulators/linalg/bda/WellContributions.cpp:116:136: error: no ‘void Opm::WellContributions::getParams(unsigned int*, unsigned int*, unsigned int*, unsigned int*)’ member function declared in class ‘Opm::WellContributions’
 void WellContributions::getParams(unsigned int *num_blocks_, unsigned int *num_std_wells_, unsigned int *dim_, unsigned int *dim_wells_){
```